_posts/2024-09-08-openloco-v24.09.1.md

```diff
@@ -0,0 +1,55 @@
+
+For a complete summary of changes, please find the
+[changelog](https://github.com/OpenLoco/OpenLoco/releases/tag/v24.09.1) on GitHub.
+
+## Randomly generated landscapes could lead to a crash (#2615)
+
+In release notes for the previous release, we proudly proclaimed the last functions of the map generator
+[had been implemented](/news/2024/09/openloco-v24.09.html#implement-map-generator-terrain-in-random-areas-2594).
+While we had tested these functions extensively using the scenario editor, we had forgotten there was
+a second path into these functions in the form of scenarios with a randomly generated landscape.
+
+As it turns out, as a result of a change earlier this month, this this code path was using invalid
+height map coordinates. The Locomotion map generator was using 512x512 height maps internally,
+presumably the intended map size before ultimately settling on the 384x384 we're all familiar with.
+When we scaled down the height map earlier this month, aiming for a slight optimisation,
+we had forgotten about this code path, leading to invalid coordinates being used --
+indeed, anything between 384 and 512. Oops.
+
+Hang on, so we think the map size was intended to be 512x512? Does that mean we can increase the map sizes
+even within the SC5/SV5 scenario/save format? Well, we [think so](https://github.com/OpenLoco/OpenLoco/issues/2390),
+but we will need to implement more code paths before we can do so. It's also questionable if it's worth
+doing. The thinking is that, once these code paths are reimplemented, the path to a custom save format
+becomes feasible as well. Ultimately, that would be the big one, potentially leading to maps much,
+much bigger than 512x512 tiles.
+
+## Reversed car components displayed incorrectly in vehicle window (#2618)
+
+Some keen eyes spotted that vehicles with some of their car components reversed were not being displayed
+properly in the vehicle window's component tab. This was the case for trains with engines on the
+front and back, for example. @duncanspumpkin quickly realised what was wrong, and implemented a fix!
+
+![image](https://github.com/user-attachments/assets/efa29970-1bda-4b9c-8392-77cae14f61fd)
+
+## Level crossings display incorrectly when they are over track, trams and road (#2619)
+
+In the previous release, @duncanspumpkin tackled the painting of road tile elements. These represented quite a
+[large part](https://openloco.io/news/2024/09/openloco-v24.09.html#implement-paint-road-2580) of the Locomotion
+code base, so we did our best to ensure no regressions. Despite this, a funny bug sneaked in:
+level crossings would not display correctly for the specific combination of a track, road, *and* tram* track!
+The resulting bug is only a visual bug, which made it even funnier: the level crossing barriers were seemingly
+ignored. Having authored the road painting, @duncanspumpkin realised what was wrong, and fixed the issue.
```

Comment:
The issue has actually been there for ~2 releases it was just chance that it was noticed after doing the paint functions